src/material-date-fns-adapter/adapter/date-fns-adapter.ts

```diff
@@ -54,7 +54,7 @@ const DAY_OF_WEEK_FORMATS = {
 export class DateFnsAdapter extends DateAdapter<Date, Locale> {
   constructor(@Optional() @Inject(MAT_DATE_LOCALE) matDateLocale: {}) {
     super();
-    super.setLocale(matDateLocale);
+    this.setLocale(matDateLocale);
```

Comment:
This change isn't part of the fix, but I figured that it's less error-prone if we were to add a `setLocale` override.